clients/src/test/java/org/apache/kafka/clients/consumer/internals/ConsumerMembershipManagerTest.java

```diff
@@ -1456,6 +1457,7 @@ public void testReconcilePartitionsRevokedWithSuccessfulAutoCommitNoCallbacks()
 
         // Complete commit request
         commitResult.complete(null);
+        verify(subscriptionState, times(2)).markPendingRevocation(Set.of(new TopicPartition("topic1", 0)));
```

Comment:
could we extend this test to ensure that the markPending happens before the commit is triggered? (Maybe using InOrder like I'm doing here)

https://github.com/apache/kafka/blob/64d020a44b49ac53b866ac88e360ba961a2730a1/clients/src/test/java/org/apache/kafka/clients/consumer/internals/ConsumerHeartbeatRequestManagerTest.java#L434-L436

Reply:
Updated it. Thanks for the suggestion.